eclipse-collections/src/main/java/org/eclipse/collections/impl/set/immutable/ImmutableUnifiedSet.java

```diff
@@ -43,6 +44,16 @@ public int size()
     @Override
     public boolean equals(Object other)
     {
+        if (other == this)
+        {
+            return true;
+        }
+
+        if (!(other instanceof Set))
+        {
+            return false;
+        }
+
```

Comment:
While this is fine, the delegate is a Set and is going to immediately perform the same check (whether Other is a Set)

Reply:
right! a bit of over-engineering ^^ thx!